evalml/data_checks/target_leakage_data_check.py

```diff
@@ -16,7 +15,7 @@ class TargetLeakageDataCheck(DataCheck):
     def __init__(self, pct_corr_threshold=0.95):
```

Comment:
I think based on the discussions we had where it seemed like there could be pros/cons of both methods, it'd be nice if the method used to determine correlation was something that could be set by the user via a parameter. We could still default it to use mutual information, but allow users to use the original pearson correlation method.